app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -15,6 +15,19 @@ def smart_proxy_ids
           ids << content_source_id
           ids.uniq.compact
         end
+
+        def update_os_from_facts
+          super
+
+          # If facts causes the OS to change, our kickstart repo might not be
+          # valid anymore. Let's reset it, either to nil or a valid one
+          ks_repo = content_facet&.kickstart_repository
```

Comment:
I've never seen this before. Could you please explain the `&`?

Reply:
It's the safe navigation operator, http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/.  `content_facet` may be nil here. 

It's been introduced already in Katello and foreman, there's a few usages of it.  It's a replacement for most uses of Rails' `#try`.

There's other nice things in Ruby 2.3, like `#dig` on Hash. http://nithinbekal.com/posts/ruby-2-3-features/